api4/plugin.go

```diff
@@ -100,22 +100,84 @@ func installPluginFromUrl(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	downloadUrl := r.URL.Query().Get("plugin_download_url")
+	pluginDownloadUrl := r.URL.Query().Get(model.PLUGIN_URL_STRING)
+	signatureDownloadUrl := r.URL.Query().Get(model.SIGNATURE_URL_STRING)
+	downloadUrls := map[string]string{
+		model.PLUGIN_URL_STRING: pluginDownloadUrl,
+	}
+	if signatureDownloadUrl != "" {
+		downloadUrls[model.SIGNATURE_URL_STRING] = signatureDownloadUrl
+	}
+
+	downloadedFiles := downloadFiles(c, downloadUrls)
+	pluginFile := downloadedFiles[model.PLUGIN_URL_STRING]
+	if pluginFile == nil {
+		return
```

Comment:
It should set the context error before returning like: 
`		c.Err = model.NewAppError("installPluginFromUrl", "api.plugin.install.invalid_url.app_error", nil, "", http.StatusBadRequest)`